reqhandler/request_handler.go

```diff
@@ -21,3 +22,34 @@ func readRequest(udpConn net.PacketConn) ([]byte, error) {
 	_, _, err := udpConn.ReadFrom(b)
 	return b, err
 }
+
+func extractID(d []byte) uint16 {
+	return (uint16(d[0]) << 8) | uint16(d[1])
```

Comment:
What is the expected behavior if []byte does not have the length this function requires? The same for line 31 in extractQuery?